boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,162 @@
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.boot.microservices;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.*;
+
+import org.apache.log4j.Logger;
+
+
+public enum IAMStarter {
+
+    //singleton instance of IAMStarter
+    INSTANCE;
+
+    private static final Logger LOGGER = Logger.getLogger(IAMStarter.class);
+
+    private static final String SEPARATOR = File.separator;
+
+    private static final String MICROSERVICE_NAME = "iam.war";
+
+    private static final String READY_MARKER = "Ready to process requests";
+
+    private static final int TIMEOUT = 180;
+
+    private static final String[] JVM_ARGS = {};
+
+    private static Process process;
+
+    private static List<String> command = new ArrayList<>();
+
+    private static boolean started = false;
+
+    /**
+     *  Start IAM microservice
+     */
+    public static Process start(String paHome, String microservicesPath)
+            throws InterruptedException, IOException, ExecutionException {
+
+        if (!started && buildJavaCommand(paHome) && buildMicroservicePath(paHome, microservicesPath)) {
+
+            LOGGER.info("Starting IAM microservice");
+
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+
+            process = processBuilder.start();
+            LOGGER.info(streamOutput(process.getInputStream()));
+
+            started = true;
+        }
+        return process;
+    }
+
+    /**
+     * Stream microservice output (using an executor) to check that it starts properly
+     */
+    private static String streamOutput(InputStream inputStream)
+            throws IOException, InterruptedException, ExecutionException {
+
+        // Stream microservice output
+        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<String> future = executor.submit(new Callable() {
+
+            public String call() throws IOException {
+
+                try {
+                    String line = null;
+
+                    while ((line = br.readLine()) != null) {
+                        //System.out.print(".");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)